odl/tomo/operators/ray_trafo.py

```diff
@@ -164,16 +165,23 @@ def __init__(self, discr_domain, geometry, **kwargs):
             range_uspace = FunctionSpace(geometry.params,
                                          out_dtype=dtype)
 
-            # Approximate cell volume
-            # TODO: angles and detector must be handled separately. While the
-            # detector should be uniformly discretized, the angles do not have
-            # to and often are not.
-            extent = float(geometry.partition.extent.prod())
-            size = float(geometry.partition.size)
-            weight = extent / size
+            if isinstance(discr_domain.weighting, NoWeighting):
+                weighting = 1.0
+            elif (isinstance(discr_domain.weighting, ConstWeighting) and
+                  np.isclose(discr_domain.weighting.const,
+                             discr_domain.cell_volume)):
+                # Approximate cell volume
+                # TODO: angles and detector must be handled separately. While the
+                # detector should be uniformly discretized, the angles do not have
+                # to and often are not.
+                extent = float(geometry.partition.extent.prod())
+                size = float(geometry.partition.size)
+                weighting = extent / size
+            else:
+                raise ValueError('unknown weighting of domain')
```

Comment:
Frankly, no. This case currently cannot happen, so I'll change it to `NotImplementedError` and leave it at that.